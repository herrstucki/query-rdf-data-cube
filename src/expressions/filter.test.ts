import { literal, namedNode } from "@rdfjs/data-model";
import DataSet from "../dataset";
import Dimension from "./dimension";
import Operator from "./operator";

const dataset: DataSet = new DataSet(
  "https://ld.stadt-zuerich.ch/query",
  {
    dataSetIri: namedNode("https://ld.stadt-zuerich.ch/statistics/dataset/BES-RAUM-ZEIT-BTA-SEX"),
    dataSetLabel: literal("BeschÃ¤ftigte nach Betriebsart, Raum, Geschlecht, Zeit"),
    graphIri: namedNode("https://linked.opendata.swiss/graph/zh/statistics"),
  },
);

test("basic", async () => {
  const a: any = new Dimension({ label: "aaaa", iri: "http://aaaa.aaa" });
  const b: any = new Dimension({ label: "bbbb", iri: "http://bbbb.bbb" });

  expect(await dataset.query().select({ a, b }).filter(a.not.bound()).applyFilters()).toMatchSnapshot();

  expect(
    await dataset.query()
    .select({ a, b })
    .filter(a.gte(b))
    .applyFilters(),
  ).toMatchSnapshot();

  expect(
    await dataset.query()
    .select({ a, b })
    .filter(a.not.gte(literal("10")))
    .applyFilters(),
  ).toMatchSnapshot();

  expect(
    await dataset.query()
    .select({ a, b })
    .filter(a.not.in([namedNode("http://example.com"), literal("foo", "en")]))
    .applyFilters(),
  ).toMatchSnapshot();
});
