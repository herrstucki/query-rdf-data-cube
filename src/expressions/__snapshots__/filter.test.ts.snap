// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic 1`] = `
Object {
  "expression": Object {
    "args": Array [
      Object {
        "args": Array [
          Object {
            "args": Array [
              Variable {
                "value": "a",
              },
            ],
            "operator": "bound",
            "type": "operation",
          },
        ],
        "operator": "!",
        "type": "operation",
      },
    ],
    "operator": "&&",
    "type": "operation",
  },
  "type": "filter",
}
`;

exports[`basic 2`] = `
Object {
  "expression": Object {
    "args": Array [
      Object {
        "args": Array [
          Variable {
            "value": "a",
          },
          Variable {
            "value": "b",
          },
        ],
        "operator": ">=",
        "type": "operation",
      },
    ],
    "operator": "&&",
    "type": "operation",
  },
  "type": "filter",
}
`;

exports[`basic 3`] = `
Object {
  "expression": Object {
    "args": Array [
      Object {
        "args": Array [
          Object {
            "args": Array [
              Variable {
                "value": "a",
              },
              Literal {
                "datatype": NamedNode {
                  "value": "http://www.w3.org/2001/XMLSchema#string",
                },
                "language": "",
                "value": "10",
              },
            ],
            "operator": ">=",
            "type": "operation",
          },
        ],
        "operator": "!",
        "type": "operation",
      },
    ],
    "operator": "&&",
    "type": "operation",
  },
  "type": "filter",
}
`;

exports[`basic 4`] = `
Object {
  "expression": Object {
    "args": Array [
      Object {
        "args": Array [
          Object {
            "args": Array [
              Variable {
                "value": "a",
              },
              NamedNode {
                "value": "http://example.com",
              },
              Literal {
                "datatype": NamedNode {
                  "value": "http://www.w3.org/1999/02/22-rdf-syntax-ns#langString",
                },
                "language": "en",
                "value": "foo",
              },
            ],
            "operator": "in",
            "type": "operation",
          },
        ],
        "operator": "!",
        "type": "operation",
      },
    ],
    "operator": "&&",
    "type": "operation",
  },
  "type": "filter",
}
`;
